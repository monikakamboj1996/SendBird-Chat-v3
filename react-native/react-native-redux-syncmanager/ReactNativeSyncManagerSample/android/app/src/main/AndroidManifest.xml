<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.reactnativesyncmanagersample">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <permission
        android:name="${applicationId}.permission.C2D_MESSAGE"
        android:protectionLevel="signature" />
    <uses-permission android:name="${applicationId}.permission.C2D_MESSAGE" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_icon"
        android:resource="@mipmap/ic_notification" />
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_channel_id"
        android:value="@string/default_notification_channel_id"/>
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:launchMode="singleTop"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

      <service android:name="io.invertase.firebase.messaging.RNFirebaseMessagingService">
        <intent-filter>
          <action android:name="com.google.firebase.MESSAGING_EVENT" />
        </intent-filter>
      </service>
      <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" />
    </application>

</manifest>
