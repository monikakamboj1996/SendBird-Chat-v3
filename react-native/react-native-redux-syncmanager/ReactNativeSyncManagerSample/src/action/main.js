
const Action = {
	Type: {
		CHANNEL_ADD: 'channel-add',
		CHANNEL_UPDATE: 'channel-update',
		CHANNEL_REMOVE: 'channel-remove',
		CHANNEL_MOVE: 'channel-move',
		CHANNEL_CLEAR: 'channel-clear'
	}
};
Action.addChannels = channels => {
	return {
		type: Action.Type.CHANNEL_ADD,
		payload: {
			channels
		}
	};
};
Action.updateChannels = channels => {
	return {
		type: Action.Type.CHANNEL_UPDATE,
		payload: {
			channels
		}
	};
}
Action.removeChannels = channels => {
	return {
		type: Action.Type.CHANNEL_REMOVE,
		payload: {
			channels
		}
	};
};
Action.moveChannels = channels => {
	return {
		type: Action.Type.CHANNEL_MOVE,
		payload: {
			channels
		}
	}
};
Action.clearChannels = () => {
	return {
		type: Action.Type.CHANNEL_CLEAR
	}
};

export default Action;