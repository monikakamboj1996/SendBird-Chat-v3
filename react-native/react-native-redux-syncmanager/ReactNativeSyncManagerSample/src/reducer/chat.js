import Action from '../action/chat';
import { findMessageIndex, getMessageIndex } from '../utils';

const _initialState = {
  channel: null,
  messages: [],
  inputMode: 'chat'
};

export const reducer = (state = _initialState, action) => {
  const newState = { ...state };
  switch (action.type) {
    case Action.Type.MESSAGE_SELECT: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < newState.messages.length; i++) {
        newState.messages[i]._selected = newState.messages[i].isIdentical(action.payload.message);
      }
      break;
    }
    case Action.Type.MESSAGE_DESELECT_ALL: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < newState.messages.length; i++) {
        newState.messages[i]._selected = false;
      }
      break;
    }
    case Action.Type.MESSAGE_ADD: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < action.payload.messages.length; i++) {
        const message = action.payload.messages[i];
        const isFailed = message.messageId === 0 && message.requestState === 'failed';
        const index = findMessageIndex(message, newState.messages, isFailed);
        if (index >= 0) {
          newState.messages.splice(index, 0, message);
          if (index > 0) {
            newState.messages[index - 1]._isPreviousMessageSentBySameUser =
              newState.messages[index - 1].sender &&
              newState.messages[index].sender &&
              newState.messages[index - 1].sender.userId === newState.messages[index].sender.userId;
          }
          if (index < newState.messages.length - 1) {
            newState.messages[index]._isPreviousMessageSentBySameUser =
              newState.messages[index].sender &&
              newState.messages[index + 1].sender &&
              newState.messages[index].sender.userId === newState.messages[index + 1].sender.userId;
          }
        }
      }
      break;
    }
    case Action.Type.MESSAGE_UPDATE: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < action.payload.messages.length; i++) {
        const message = action.payload.messages[i];
        const index = getMessageIndex(message, newState.messages);
        if (index >= 0) {
          message._isPreviousMessageSentBySameUser = newState.messages[index]._isPreviousMessageSentBySameUser;
          newState.messages[index] = message;
          if (action.payload.isTransformFailedToManual) {
            newState.messages[index]._isManualResend = true;
          }
        }
      }
      break;
    }
    case Action.Type.MESSAGE_REMOVE: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < action.payload.messages.length; i++) {
        const message = action.payload.messages[i];
        const index = getMessageIndex(message, newState.messages);
        if (index >= 0) {
          newState.messages.splice(index, 1);
          if (index < newState.messages.length - 1) {
            newState.messages[index]._isPreviousMessageSentBySameUser =
              newState.messages[index + 1].sender &&
              newState.messages[index].sender &&
              newState.messages[index + 1].sender.userId === newState.messages[index].sender.userId;
          }
        }
      }
      break;
    }
    case Action.Type.MESSAGE_CLEAR: {
      newState.messages = [];
      break;
    }
  }
  return newState;
};
