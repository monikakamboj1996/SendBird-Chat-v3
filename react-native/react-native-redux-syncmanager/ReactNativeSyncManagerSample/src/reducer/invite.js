import Action from "../action/invite";

const _initialState = {
	users: [],
	err: null
};

export const reducer = (state = _initialState, action) => {
	const newState = { ...state };
	switch(action.type) {
		case Action.Type.USER_FETCH: {
			newState.users = newState.users.concat(action.payload.users);
			break;
		}
	}
	return newState;
};