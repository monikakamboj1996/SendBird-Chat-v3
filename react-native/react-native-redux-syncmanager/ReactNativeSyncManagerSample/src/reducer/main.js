import Action from '../action/main';
import { findChannelIndex, getChannelIndex } from '../utils';

const _initialState = {
  channels: []
};

export const reducer = (state = _initialState, action) => {
  const newState = { ...state };
  switch (action.type) {
    case Action.Type.CHANNEL_ADD: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = findChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels.splice(index, 0, channel);
        }
      }
      break;
    }
    case Action.Type.CHANNEL_UPDATE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = getChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels[index] = channel;
        }
      }
      break;
    }
    case Action.Type.CHANNEL_REMOVE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = getChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels.splice(index, 1);
        }
      }
      break;
    }
    case Action.Type.CHANNEL_MOVE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const from = getChannelIndex(channel, newState.channels);
        if (from >= 0) {
          newState.channels.splice(from, 1);
        }
        const to = findChannelIndex(channel, newState.channels);
        newState.channels.splice(to, 0, channel);
      }
      break;
    }
    case Action.Type.CHANNEL_CLEAR: {
      newState.channels = [];
      break;
    }
  }
  return newState;
};
