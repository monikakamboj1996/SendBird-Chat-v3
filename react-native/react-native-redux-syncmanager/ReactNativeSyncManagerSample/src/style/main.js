
import { StyleSheet } from 'react-native';

export const style = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#f3f3f3'
	},
	actionButtonShadow: {
		shadowColor: "#333",
		shadowOpacity: 0.6,
		shadowOffset: {
			height: 1,
			width: 1
		}
	},
	actionButtonIcon: {
		fontSize: 20,
    height: 22,
    color: 'white'
	},
	signoutButtonContainer: {
		marginLeft: 0,
		marginRight: 10
	},
	signoutButton: {
		backgroundColor: 'transparent',
		paddingHorizontal: 10
	}
});