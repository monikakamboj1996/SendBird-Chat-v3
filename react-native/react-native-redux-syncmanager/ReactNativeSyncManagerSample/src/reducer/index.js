import { combineReducers } from 'redux';
import { reducer as main } from './main';
import { reducer as chat } from './chat';
import { reducer as invite } from './invite';
import { reducer as member } from './member';
import { reducer as signin } from './signin';

export default combineReducers({
  main,
  chat,
  invite,
  member,
  signin
});
