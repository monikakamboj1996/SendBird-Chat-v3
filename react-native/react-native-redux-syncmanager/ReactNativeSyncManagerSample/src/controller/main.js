import React from 'react';
import { Alert, Platform, View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';

import AsyncStorage from '@react-native-community/async-storage';
import { FlatList } from 'react-native-gesture-handler';
import ActionButton from 'react-native-action-button';
import Icon from 'react-native-vector-icons/Ionicons';
import firebase from 'react-native-firebase';

import SendBird from 'sendbird';
import SendBirdSyncManager from 'sendbird-syncmanager';

import { style } from '../style/main';
import Action from '../action/main';
import GroupChannelView from '../view/groupChannel';
import { navigator } from '../navigator';

export class MainController extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'SendBird',
      headerRight: (
        <Button
          containerStyle={style.signoutButtonContainer}
          buttonStyle={style.signoutButton}
          icon={<Icon name="md-log-out" color="#7d62d9" size={28}></Icon>}
          onPress={() => {
            Alert.alert('Sign out', 'Do you want to sign out?', [
              { text: 'Cancel' },
              {
                text: 'OK',
                onPress: () => {
                  AsyncStorage.removeItem('currentUser', err => {
                    SendBirdSyncManager.getInstance().reset();
                    const sb = SendBird.getInstance();
                    firebase
                      .messaging()
                      .getToken()
                      .then(fcmToken => {
                        if (fcmToken) {
                          if (Platform.OS === 'ios') {
                            sb.unregisterAPNSPushTokenForCurrentUser(fcmToken, (result, err) => {
                              sb.disconnect(() => {
                                navigation.replace('Signin', {});
                              });
                            });
                          } else {
                            sb.unregisterGCMPushTokenForCurrentUser(fcmToken, (result, err) => {
                              sb.disconnect(() => {
                                navigation.replace('Signin', {});
                              });
                            });
                          }
                        } else {
                          sb.disconnect(() => {
                            navigation.replace('Signin', {});
                          });
                        }
                      });
                  });
                }
              }
            ]);
          }}
        />
      )
    };
  };
  static getDerivedStateFromProps(props, state) {
    state.channels = props.channels;
    return state;
  }
  constructor(props) {
    super(props);
    this.collection = null;
    this.state = {
      channels: []
    };
  }
  componentDidMount() {
    const { navigation } = this.props;
    const userFromParams = navigation.getParam('user', null);
    if (!userFromParams) {
      AsyncStorage.getItem('currentUser', (err, result) => {
        if (result) {
          const user = JSON.parse(result);
          this.init(user.userId);
        } else {
          this.requireLogin();
        }
      });
    } else {
      this.init(userFromParams.userId);
    }
  }
  componentWillUnmount() {
    if (this.onTokenRefreshListener) {
      this.onTokenRefreshListener();
    }
    if (this.onPushNotificationListener) {
      this.onPushNotificationListener();
    }
    if (this.onPushNotificationOpened) {
      this.onPushNotificationOpened();
    }
    this.props.clearChannels();
    const manager = SendBirdSyncManager.getInstance();
    if (manager) {
      manager.clearCache();
    }
    if (this.collection) {
      this.collection = null;
    }
  }
  init(userId) {
    const sb = SendBird.getInstance();
    sb.connect(userId, (_, err) => {
      if (!err) {
        firebase
          .messaging()
          .getToken()
          .then(fcmToken => {
            if (fcmToken) {
              this.registerPushToken(fcmToken);
            }
          });
      }
    });

    firebase
      .messaging()
      .hasPermission()
      .then(enabled => {
        if (!enabled) {
          firebase
            .messaging()
            .requestPermission()
            .then(() => {
              this.addPushNotificationListener();
            })
            .catch(err => {
              console.log('Failed to get permission', err);
            });
        } else {
          this.addPushNotificationListener();
        }
      });
    this.addPushTokenRefreshListener();

    this.onPushNotificationOpened = firebase.notifications().onNotificationOpened(notificationOpen => {
      firebase.notifications().removeAllDeliveredNotifications();
      if (notificationOpen) {
        const notification = notificationOpen.notification;
        if (notification.data && notification.data.channel && notification.data.channel.channel_url) {
          sb.GroupChannel.getChannel(notification.data.channel.channel_url, (channel, err) => {
            if (!err) {
              this.openChat(channel);
            }
          });
        }
      }
    });
    firebase.notifications().removeAllDeliveredNotifications();
    const connectionHandler = new sb.ConnectionHandler();
    connectionHandler.onReconnectFailed = () => {
      sb.reconnect();
    };
    connectionHandler.onReconnectStarted = () => {};
    connectionHandler.onReconnectSucceeded = () => {
      const manager = SendBirdSyncManager.getInstance();
      manager.resumeSync();
    };
    sb.addConnectionHandler('connection handler', connectionHandler);

    const options = new SendBirdSyncManager.Options();
    options.messageCollectionCapacity = 2000;
    options.messageResendPolicy = 'automatic';
    options.failedMessageRetentionDays = 7;
    options.maxFailedMessageCountPerChannel = 50;
    options.automaticMessageResendRetryCount = 4;
    SendBirdSyncManager.setup(userId, options, err => {
      if (!err) {
        firebase
          .notifications()
          .getInitialNotification()
          .then(notificationOpen => {
            if (notificationOpen) {
              const notification = notificationOpen.notification;
              if (notification.data && notification.data.channel && notification.data.channel.channel_url) {
                sb.GroupChannel.getChannel(notification.data.channel.channel_url, (channel, err) => {
                  if (!err) {
                    this.openChat(channel);
                  }
                });
              }
            }
          });

        const query = sb.GroupChannel.createMyGroupChannelListQuery();
        query.limit = 50;
        query.includeEmpty = false;
        query.order = 'latest_last_message';

        this.collection = new SendBirdSyncManager.ChannelCollection(query);
        const collectionHandler = new SendBirdSyncManager.ChannelCollection.CollectionHandler();
        collectionHandler.onChannelEvent = (action, channels) => {
          switch (action) {
            case 'insert': {
              this.props.addChannels(channels);
              break;
            }
            case 'update': {
              this.props.updateChannels(channels);
              break;
            }
            case 'move': {
              this.props.moveChannels(channels);
              break;
            }
            case 'remove': {
              this.props.removeChannels(channels);
              break;
            }
            case 'clear': {
              this.props.clearChannels();
              break;
            }
          }
        };
        this.collection.setCollectionHandler(collectionHandler);
        this.next();
      } else {
        console.log(err);
      }
    });
  }
  addPushTokenRefreshListener() {
    this.onTokenRefreshListener = firebase.messaging().onTokenRefresh(fcmToken => {
      this.registerPushToken(fcmToken);
    });
  }
  addPushNotificationListener() {
    this.onPushNotificationListener = firebase.messaging().onMessage(message => {
      this.handlePushNotification(message);
    });
  }
  handlePushNotification(message) {
    // DO NOTHING FOR FOREGROUND PUSH NOTIFICATION
  }
  registerPushToken(fcmToken) {
    const sb = SendBird.getInstance();
    if (Platform.OS === 'ios') {
      // WARNING! FCM token doesn't work in request to APNs.
      // Use APNs token here instead.
      firebase
        .messaging()
        .ios.getAPNSToken()
        .then(token => {
          sb.registerAPNSPushTokenForCurrentUser(token, (result, err) => {
            if (err) {
              console.log(err);
            }
          });
        })
        .catch(error => {
          console.error(error);
        });
    } else {
      sb.registerGCMPushTokenForCurrentUser(fcmToken, (result, err) => {
        if (err) {
          console.log(err);
        }
      });
    }
  }
  requireLogin() {
    this.props.navigation.replace('Signin', {});
  }
  createChannel() {
    this.props.navigation.push('Invite', {});
  }
  openChat(channel) {
    if (navigator.top() !== channel.url) {
      this.props.navigation.push('Chat', { channel });
    }
  }
  next() {
    if (this.collection) {
      this.collection.fetch(err => {
        if (err) {
          console.log(err);
        }
      });
    }
  }
  render() {
    return (
      <View style={style.container}>
        <FlatList
          data={this.state.channels}
          keyExtractor={channel => channel.url}
          onEndReached={this.next.bind(this)}
          renderItem={bundle => {
            const channel = bundle.item;
            return <GroupChannelView channel={channel} onPress={this.openChat.bind(this)}></GroupChannelView>;
          }}
        />
        <ActionButton
          buttonColor="#6e5baa"
          degrees={0}
          shadowStyle={style.actionButtonShadow}
          fixNativeFeedbackRadius={true}
          onPress={this.createChannel.bind(this)}
        >
          <Icon name="md-create" style={style.actionButtonIcon} />
        </ActionButton>
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { main } = previousState;
    return { ...main };
  },
  Action
)(MainController);
