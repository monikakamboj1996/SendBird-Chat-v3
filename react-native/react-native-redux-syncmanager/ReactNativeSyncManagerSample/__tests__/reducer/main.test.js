import { reducer } from '../../src/reducer/main';
import Action from '../../src/action/main';

describe('reducer/main', () => {
  test('checking initial state', () => {
    const action = { type: 'nonexistent' };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  test('CHANNEL_ADD initially empty', () => {
    const action = {
      type: Action.Type.CHANNEL_ADD,
      payload: { channels: [{ url: 'url 1' }, { url: 'url 2' }] }
    };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  test('CHANNEL_ADD initially non-empty, insert by createdAt', () => {
    const action = {
      type: Action.Type.CHANNEL_ADD,
      payload: { channels: [{ createdAt: 4, url: '4' }, { createdAt: 6, url: '6' }] }
    };
    const initState = {
      channels: [
        {
          createdAt: 5,
          url: '5'
        },
        {
          createdAt: 3,
          url: '3'
        },
        {
          createdAt: 2,
          url: '2'
        },
        {
          createdAt: 1,
          url: '1'
        }
      ]
    };
    expect(reducer(initState, action)).toMatchSnapshot();
  });

  test('CHANNEL_ADD initially non-empty, insert by lastMessage', () => {
    const action = {
      type: Action.Type.CHANNEL_ADD,
      payload: { channels: [{ createdAt: 4, url: '4' }, { createdAt: 6, url: '6' }] }
    };
    const initState = {
      channels: [
        {
          lastMessage: {
            createdAt: 5
          },
          url: '5'
        },
        {
          lastMessage: {
            createdAt: 3
          },
          url: '3'
        },
        {
          lastMessage: {
            createdAt: 2
          },
          url: '2'
        },
        {
          lastMessage: {
            createdAt: 1
          },
          url: '1'
        }
      ]
    };
    expect(reducer(initState, action)).toMatchSnapshot();
  });

  test('CHANNEL_UPDATE', () => {
    const action = {
      type: Action.Type.CHANNEL_UPDATE,
      payload: { channels: [{ url: '1', updated: true }, { url: '2', updated: true }, { url: '6', updated: true }] }
    };
    const initState = {
      channels: [
        {
          url: '5'
        },
        {
          url: '3'
        },
        {
          url: '2'
        },
        {
          url: '1'
        }
      ]
    };
    expect(reducer(initState, action)).toMatchSnapshot();
  });

  test('CHANNEL_REMOVE', () => {
    const action = {
      type: Action.Type.CHANNEL_REMOVE,
      payload: { channels: [{ url: '1' }, { url: '2' }, { url: '6' }] }
    };
    const initState = {
      channels: [
        {
          url: '5'
        },
        {
          url: '3'
        },
        {
          url: '2'
        },
        {
          url: '1'
        }
      ]
    };
    expect(reducer(initState, action)).toMatchSnapshot();
  });

  test('CHANNEL_MOVE', () => {
    const action = {
      type: Action.Type.CHANNEL_MOVE,
      payload: { channels: [{ url: '1', createdAt: 10 }] }
    };
    const initState = {
      channels: [
        {
          url: '5',
          createdAt: 5
        },
        {
          url: '3',
          createdAt: 3
        },
        {
          url: '2',
          createdAt: 2
        },
        {
          url: '1',
          createdAt: 1
        }
      ]
    };
    expect(reducer(initState, action)).toMatchSnapshot();
  });

  test('CHANNEL_CLEAR', () => {
    const action = {
      type: Action.Type.CHANNEL_CLEAR
    };
    const initState = {
      channels: [
        {
          url: '5'
        },
        {
          url: '3'
        },
        {
          url: '2'
        },
        {
          url: '1'
        }
      ]
    };
    expect(reducer(initState, action)).toMatchSnapshot();
  });
});
