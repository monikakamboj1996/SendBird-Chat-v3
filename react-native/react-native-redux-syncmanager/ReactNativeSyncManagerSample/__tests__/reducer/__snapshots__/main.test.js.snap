// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer/main CHANNEL_ADD initially empty 1`] = `
Object {
  "channels": Array [
    Object {
      "url": "url 1",
    },
    Object {
      "url": "url 2",
    },
  ],
}
`;

exports[`reducer/main CHANNEL_ADD initially non-empty, insert by createdAt 1`] = `
Object {
  "channels": Array [
    Object {
      "createdAt": 6,
      "url": "6",
    },
    Object {
      "createdAt": 5,
      "url": "5",
    },
    Object {
      "createdAt": 4,
      "url": "4",
    },
    Object {
      "createdAt": 3,
      "url": "3",
    },
    Object {
      "createdAt": 2,
      "url": "2",
    },
    Object {
      "createdAt": 1,
      "url": "1",
    },
  ],
}
`;

exports[`reducer/main CHANNEL_ADD initially non-empty, insert by lastMessage 1`] = `
Object {
  "channels": Array [
    Object {
      "createdAt": 6,
      "url": "6",
    },
    Object {
      "lastMessage": Object {
        "createdAt": 5,
      },
      "url": "5",
    },
    Object {
      "createdAt": 4,
      "url": "4",
    },
    Object {
      "lastMessage": Object {
        "createdAt": 3,
      },
      "url": "3",
    },
    Object {
      "lastMessage": Object {
        "createdAt": 2,
      },
      "url": "2",
    },
    Object {
      "lastMessage": Object {
        "createdAt": 1,
      },
      "url": "1",
    },
  ],
}
`;

exports[`reducer/main CHANNEL_CLEAR 1`] = `
Object {
  "channels": Array [],
}
`;

exports[`reducer/main CHANNEL_MOVE 1`] = `
Object {
  "channels": Array [
    Object {
      "createdAt": 10,
      "url": "1",
    },
    Object {
      "createdAt": 5,
      "url": "5",
    },
    Object {
      "createdAt": 3,
      "url": "3",
    },
    Object {
      "createdAt": 2,
      "url": "2",
    },
  ],
}
`;

exports[`reducer/main CHANNEL_REMOVE 1`] = `
Object {
  "channels": Array [
    Object {
      "url": "5",
    },
    Object {
      "url": "3",
    },
  ],
}
`;

exports[`reducer/main CHANNEL_UPDATE 1`] = `
Object {
  "channels": Array [
    Object {
      "url": "5",
    },
    Object {
      "url": "3",
    },
    Object {
      "updated": true,
      "url": "2",
    },
    Object {
      "updated": true,
      "url": "1",
    },
  ],
}
`;

exports[`reducer/main checking initial state 1`] = `
Object {
  "channels": Array [],
}
`;
